bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVConnectionManager.java

```diff
@@ -0,0 +1,1383 @@
+        logger.debug("{} - Command sender thread started {}", handler.getThingID(), config.port);
+        try {
+            while (!Thread.currentThread().isInterrupted() && writer != null) {
+                GoogleTVCommand command = sendQueue.take();
+
+                try {
+                    BufferedWriter writer = this.writer;
+                    if (writer != null) {
+                        logger.trace("{} - Raw GoogleTV command decodes as: {}", handler.getThingID(),
+                                GoogleTVRequest.decodeMessage(command.toString()));
+                        writer.write(command.toString());
+                        writer.flush();
+                    }
+                } catch (InterruptedIOException e) {
+                    logger.debug("Interrupted while sending to GoogleTV");
+                    setStatus(false, "Interrupted");
+                    break; // exit loop and terminate thread
+                } catch (IOException e) {
+                    logger.warn("{} - Communication error, will try to reconnect GoogleTV. Error: {}",
+                            handler.getThingID(), e.getMessage());
+                    setStatus(false, "Communication error, will try to reconnect");
+                    sendQueue.add(command); // Requeue command
+                    this.isLoggedIn = false;
+                    reconnect();
+                    break; // reconnect() will start a new thread; terminate this one
+                }
+                if (config.delay > 0) {
+                    Thread.sleep(config.delay); // introduce delay to throttle send rate
+                }
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } finally {
+            logger.debug("{} - Command sender thread exiting {}", handler.getThingID(), config.port);
+        }
+    }
+
+    private void shimSenderThreadJob() {
+        logger.debug("Shim sender thread started");
+        try {
+            while (!Thread.currentThread().isInterrupted() && shimWriter != null) {
+                GoogleTVCommand command = shimQueue.take();
+
+                try {
+                    BufferedWriter writer = this.shimWriter;
+                    if (writer != null) {
+                        logger.trace("Shim received from google: {}",
+                                GoogleTVRequest.decodeMessage(command.toString()));
+                        writer.write(command.toString());
+                        writer.flush();
+                    }
+                } catch (InterruptedIOException e) {
+                    logger.debug("Shim interrupted while sending.");
+                    break; // exit loop and terminate thread
+                } catch (IOException e) {
+                    logger.warn("Shim communication error. Error: {}", e.getMessage());
+                    break; // reconnect() will start a new thread; terminate this one
+                }
```

Comment:
Can these also be moved to the try-catch outside the loop?

Reply:
I don't see why we can't, but as this is part of shim it's almost never going to actually.  Also keeping it inside the loop keeps it the same as above which makes it easier for me to debug.